rust-runtime/aws-smithy-http-server/src/routing/request_spec.rs

```diff
@@ -107,14 +107,13 @@ impl From<&PathSpec> for Regex {
                 .map(|segment_spec| match segment_spec {
                     PathSegment::Literal(literal) => literal,
                     // TODO(https://github.com/awslabs/smithy/issues/975) URL spec says it should be ASCII but this regex accepts UTF-8:
-                    // `*` instead of `+` because the empty string `""` can be bound to a label.
-                    PathSegment::Label => "[^/]*",
+                    PathSegment::Label => "[^/]+",
```

Comment:
Unfortunately we don't know whether this is correct. The Smithy spec should detail what to do in the case of empty path segments. We are tracking this in https://github.com/awslabs/smithy/issues/1024. Feel free to share your thoughts there.

For the moment we're allowing empty path segments, since otherwise the empty string `""` can never be bound to a label. Note that this is also the approach we've taken when doing label deserialization (note it's [`take_until`](https://docs.rs/nom/latest/nom/bytes/complete/fn.take_until.html) and not [`take_until1`](https://docs.rs/nom/latest/nom/bytes/complete/fn.take_until1.html)):

https://github.com/awslabs/smithy-rs/blob/main/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt#L639

However, we're aware that this is at odds with the S3 model https://github.com/awslabs/smithy/issues/1024#issuecomment-1012066749.

Reply:
Got it. Let me take another look.